api/src/main/java/org/openmrs/module/eptsreports/reporting/reports/SetupEc20DataQualityReport.java

```diff
@@ -0,0 +1,135 @@
+package org.openmrs.module.eptsreports.reporting.reports;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Properties;
+import org.openmrs.Location;
+import org.openmrs.module.eptsreports.metadata.HivMetadata;
+import org.openmrs.module.eptsreports.reporting.library.cohorts.data.quality.SummaryEc20DataQualityCohorts;
+import org.openmrs.module.eptsreports.reporting.library.datasets.data.quality.Ec20PatientListDataset;
+import org.openmrs.module.eptsreports.reporting.library.datasets.data.quality.GetCustomConfigurationDataset;
+import org.openmrs.module.eptsreports.reporting.library.datasets.data.quality.SummaryEc20DataQualityDataset;
+import org.openmrs.module.eptsreports.reporting.reports.manager.EptsDataExportManager;
+import org.openmrs.module.eptsreports.reporting.utils.EptsReportUtils;
+import org.openmrs.module.reporting.ReportingConstants;
+import org.openmrs.module.reporting.ReportingException;
+import org.openmrs.module.reporting.evaluation.parameter.Mapped;
+import org.openmrs.module.reporting.evaluation.parameter.Parameter;
+import org.openmrs.module.reporting.report.ReportDesign;
+import org.openmrs.module.reporting.report.definition.ReportDefinition;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class SetupEc20DataQualityReport extends EptsDataExportManager {
+
+  private SummaryEc20DataQualityCohorts summaryEc20DataQualityCohorts;
+
+  private SummaryEc20DataQualityDataset summaryEc20DataQualityDataset;
+
+  private Ec20PatientListDataset ec20PatientListDataset;
+
+  private HivMetadata hivMetadata;
+
+  private GetCustomConfigurationDataset getCustomConfigurationDataset;
+
+  @Autowired
+  public SetupEc20DataQualityReport(
```

Comment:
@KelvinWanyama what is the reason for creating a separate report for Ec20?

Reply:
i saw this and although didn't really think about why it was a separate one, this question perfectly makes sense to me.

Reply:
The EC20 had different requirements from EC1-EC19 data quality checks. We had a discussion on the same with Pinki and @ningosi . The agreement was to have the check as a separate report.

EC1-EC19: Uses the base cohort and thus takes into account the patients enrolled in the program. For EC20, the requirements need patients not enrolled in the program (TARV): **Patients that are not enrolled in TARV but has a consultation or drugs pick up recorded in the system**

Reply:
Ok :+1: